src/services/cube-handler.ts

```diff
@@ -726,8 +867,20 @@ export const createBaseCube = async (dataset: Dataset, endRevision: Revision): P
         measureColumn
     );
 
+    if (referenceDataPresent(dataset)) {
+        await loadReferenceDataIntoCube(quack);
+    }
+
     await setupDimensions(quack, dataset, selectStatementsMap, joinStatements, orderByStatements);
 
+    if (referenceDataPresent(dataset)) {
+        await cleanUpReferenceDataTables(quack);
+        joinStatements.push(`JOIN reference_data_info ON reference_data.item_id=reference_data_info.item_id`);
+        joinStatements.push(`    AND reference_data.category_key=reference_data_info.category_key`);
+        joinStatements.push(`    AND reference_data.version_no=reference_data_info.version_no`);
+        joinStatements.push(`    AND reference_data_info.lang='#LANG#'`);
```

Comment:
This whitespace is a bit strange... is this to make it more readable in debug logs?

Reply:
Yes... When you get the debug statement for the default view its to allow picking out the reference data query part of the query.